os/include/tinyara/wqueue.h

```diff
@@ -488,6 +491,19 @@ void lpwork_boostpriority(uint8_t reqprio);
 void lpwork_restorepriority(uint8_t reqprio);
 #endif
 
+/****************************************************************************
+ * Name: work_get_current
+ *
+ * Description:
+ *   Get the current running worker
+ ****************************************************************************/
+#if defined(CONFIG_DEBUG_WORKQUEUE) && defined(__KERNEL__)
+worker_t work_get_current(void);
+
+#define IS_HPWORK (strncmp(this_task()->name, HPWORKNAME, sizeof(HPWORKNAME)) == 0)
+#define IS_LPWORK (strncmp(this_task()->name, LPWORKNAME, sizeof(HPWORKNAME)) == 0)
+#endif
```

Comment:
Even hp and lp workqueue config can be controlled separately, here both are used without the conditional.
It causes build break. To keep code simple, I suggest below.
```
#if defined(CONFIG_DEBUG_WORKQUEUE) && defined(__KERNEL__)
worker_t work_get_current(void);

#ifdef CONFIG_SCHED_HPWORK
#define IS_HPWORK (strncmp(this_task()->name, HPWORKNAME, sizeof(HPWORKNAME)) == 0)
#else
#define IS_HPWORK (false)
#endif
#ifdef CONFIG_SCHED_LPWORK
#define IS_LPWORK (strncmp(this_task()->name, LPWORKNAME, sizeof(HPWORKNAME)) == 0)
#else
#define IS_LPWORK (false)
#endif
#endif
```

Reply:
Modified. Thanks.